storage/store.go

```
@@ -538,6 +539,16 @@ func (s *Store) DeleteVault(publicKeyECDSA string) error {
 	if _, err := s.db.Exec("DELETE FROM coins WHERE public_key_ecdsa = ?", publicKeyECDSA); err != nil {
 		return fmt.Errorf("could not delete vault coins: %w", err)
 	}
+	// The ON DELETE CASCADE on these tables already covers them once foreign
+	// keys are on for every pooled connection. Deleting them explicitly keeps
+	// key material from outliving its vault even on a database whose rows
+	// predate that fix, or if the pragma ever regresses again.
+	if _, err := s.db.Exec("DELETE FROM keyshares WHERE public_key_ecdsa = ?", publicKeyECDSA); err != nil {
```

golangci-lint (2.13.2)

[error] (*database/sql.DB).Exec must not be called. use (*database/sql.DB).ExecContext

(noctx)